### PR TITLE
edit spaces and their venue values in admin v3

### PR DESCRIPTION
**Pull request by denisdimitrov**
- Adds additional input fields to the edit space form. Allows venue edition as well.
- Passes the parentId automatically to newly created child venues

### CODE REVIEW

File: src/components/molecules/SpaceEditForm/SpaceEditForm.tsx [hunk 1]
```diff
@@ -0,0 +1,484 @@
...
+    loading: isLoadingRoomVenue,
+    error: roomVenueError,
+    value: roomVenue,
+  } = useAsync(async () => {
+    if (!roomVenueId) return;
+
+    return await fetchVenue(roomVenueId);
+  }, [roomVenueId]);
+
+  console.log(roomVenue);
```
Comment (yurii-lubynets) [inline src/components/molecules/SpaceEditForm/SpaceEditForm.tsx:73]: let's remove console

File: src/components/molecules/SpaceEditForm/SpaceEditForm.tsx [hunk 2]
```diff
@@ -0,0 +1,484 @@
...
+    validationSchema: roomEditSchema,
+    defaultValues,
+  });
+
+  useEffect(() => reset(defaultValues), [defaultValues, reset]);
+
+  const values = watch("room");
+  const venueValues = watch("venue");
+
+  console.log(venueValues);
```
Comment (yurii-lubynets) [inline src/components/molecules/SpaceEditForm/SpaceEditForm.tsx:127]: let's remove console

File: src/components/molecules/SpaceEditForm/SpaceEditForm.tsx [hunk 3]
```diff
@@ -0,0 +1,484 @@
...
+    if (!user || !venueId) return;
+
+    const roomData: RoomInput = {
+      ...(room as RoomInput),
+      ...(updatedRoom as RoomInput),
+      ...values,
+    };
+
+    await upsertRoom(roomData, venueId, user, roomIndex);
+    console.log(room.template);
```
Comment (yurii-lubynets) [inline src/components/molecules/SpaceEditForm/SpaceEditForm.tsx:164]: let's remove console

Comment (denisdimitrov) [reply src/components/molecules/SpaceEditForm/SpaceEditForm.tsx]: Fixed in https://github.com/sparkletown/sparkle/pull/2312/commits/cf45b3f8b266cc84df543b4a8c34438e9b9f6938

File: src/components/molecules/SpaceEditForm/SpaceEditForm.tsx (hunk 2 above)
Comment (denisdimitrov) [reply src/components/molecules/SpaceEditForm/SpaceEditForm.tsx]: Fixed in https://github.com/sparkletown/sparkle/pull/2312/commits/cf45b3f8b266cc84df543b4a8c34438e9b9f6938

File: src/components/molecules/VenueRoomItem/VenueRoomItem.tsx [hunk 4]
```diff
@@ -80,7 +80,10 @@ export const VenueRoomItem: React.FC<VenueRoomItemProps> = ({
     // TS doesn't work properly with const statements and won't 'know' that this is already checked.
     // That's why this is inline instead of isVenuePortal
     if (template !== RoomTemplate.external) {
```
Comment (dmitrymarchuk) [inline src/components/molecules/VenueRoomItem/VenueRoomItem.tsx:82]: You can turn `isVenuePortal` to a [type-guard function](https://basarat.gitbook.io/typescript/type-system/typeguard#user-defined-type-guards) and get rid of the duplication & comment above

File: src/components/molecules/VenueRoomItem/VenueRoomItem.tsx [hunk 5]
```diff
@@ -80,7 +80,10 @@ export const VenueRoomItem: React.FC<VenueRoomItemProps> = ({
...
-      const venueData = buildEmptyVenue(roomValues.venueName, template);
+      const venueData = {
+        ...buildEmptyVenue(roomValues.venueName, template),
```
Comment (dmitrymarchuk) [inline src/components/molecules/VenueRoomItem/VenueRoomItem.tsx:84]: To retain consistency you can pass `parentId` to `buildEmptyVenue` as it seems like always adding `parentId` is a new "norm.

File: src/types/rooms.ts [hunk 6]
```diff
@@ -33,7 +33,7 @@ export interface RoomData_v2 {
   title?: string;
   subtitle?: string;
   url?: string;
-  description?: string;
+  about?: string;
```
Comment (dmitrymarchuk) [inline src/types/rooms.ts:36]: It seems that this change is not supported in other places, isn't it?

File: functions/venue.js [hunk 7]
```diff
@@ -767,6 +768,135 @@ exports.updateVenue_v2 = functions.https.onCall(async (data, context) => {
   admin.firestore().collection("venues").doc(venueId).update(updated);
 });
 
+exports.updateVenueNG = functions.https.onCall(async (data, context) => {
```
Comment (dmitrymarchuk) [inline functions/venue.js:771]: Why do we need a _third_ variant of updating venues?😃

**Review by dmitrymarchuk — CHANGES_REQUESTED** [review]
Some questions/requests

File: src/components/atoms/PortalVisibility/PortalVisibility.tsx [hunk 8]
```diff
@@ -16,14 +17,10 @@ export const labelOptions = LABEL_VISIBILITY_OPTIONS.map((option) => (
 ));
 
 export const PortalVisibility: React.FC<PortalVisibilityProps> = ({
+  name = "visibility",
   register,
 }) => (
-  <select
-    name="visibility"
-    id="visibility"
```
Comment (goran-peoski-work) [inline src/components/atoms/PortalVisibility/PortalVisibility.tsx:23]: Good catch

File: src/components/molecules/SpaceEditForm/SpaceEditForm.scss [hunk 9]
```diff
@@ -1,16 +1,30 @@
 @import "scss/constants.scss";
 
-.EditRoomForm {
+$spaces-footer-height: 75px;
```
Comment (goran-peoski-work) [inline src/components/molecules/SpaceEditForm/SpaceEditForm.scss:3]: Shouldn't this already be covered by or at least moved to this block https://github.com/sparkletown/sparkle/blob/staging/src/scss/constants.scss#L69-L82 ?

Comment (goran-peoski-work) [reply src/components/molecules/SpaceEditForm/SpaceEditForm.scss:3]: As a reference (might as well copy from): https://github.com/sparkletown/sparkle/blob/staging/src/components/organisms/AdminVenueView/components/TabFooter/TabFooter.scss#L14-L19

File: src/components/molecules/SpaceEditForm/SpaceEditForm.tsx [hunk 10]
```diff
@@ -0,0 +1,488 @@
...
+                )}
+            </>
+          )}
+
+          <Button
+            disabled={isUpdating || isDeleting}
+            onClick={deleteSelectedRoom}
+          >
+            Delete room
+          </Button>
```
Comment (goran-peoski-work) [inline src/components/molecules/SpaceEditForm/SpaceEditForm.tsx:464]: This one button can safely be updated. 
```suggestion
          <ButtonNG
            disabled={isUpdating || isDeleting}
            loading={isUpdating || isDeleting}
            onClick={deleteSelectedRoom}
            variant="danger"
          >
            Delete room
          </ButtonNG>
```



[OOS]
As for the bottom ones they should probably become part of an improved `TabFooter` one that allows for different types/numbers of buttons to accommodate design. Probably new PR under https://github.com/sparkletown/internal-sparkle-issues/issues/1195

**Review by goran-peoski-work — CHANGES_REQUESTED** [review]
Some notes on the SpaceEditForm style

File: src/components/molecules/VenueRoomItem/VenueRoomItem.tsx (hunk 4 above)
Comment (denisdimitrov) [reply src/components/molecules/VenueRoomItem/VenueRoomItem.tsx:82]: This is going to be removed in: https://github.com/sparkletown/sparkle/pull/2354/files

File: functions/venue.js (hunk 7 above)
Comment (denisdimitrov) [reply functions/venue.js:766]: The other ones are creating venue objects with incorrect data that was used for admin v2, this one is going to create an empty object add only the passed fields and then update the firestore document

File: src/types/rooms.ts (hunk 6 above)
Comment (denisdimitrov) [reply src/types/rooms.ts:42]: Description was also an admin v2 thing, and it wasn't used anywhere.

File: src/components/molecules/SpaceEditForm/SpaceEditForm.tsx (hunk 10 above)
Comment (denisdimitrov) [reply src/components/molecules/SpaceEditForm/SpaceEditForm.tsx]: Fixed in: https://github.com/sparkletown/sparkle/pull/2312/commits/0023b3b4d849d0af4ae51d5af3ca558495cb927e

File: src/components/molecules/SpaceEditForm/SpaceEditForm.scss (hunk 9 above)
Comment (denisdimitrov) [reply src/components/molecules/SpaceEditForm/SpaceEditForm.scss]: Fixed in: https://github.com/sparkletown/sparkle/pull/2312/commits/0023b3b4d849d0af4ae51d5af3ca558495cb927e

File: src/components/molecules/VenueRoomItem/VenueRoomItem.tsx (hunk 5 above)
Comment (denisdimitrov) [reply src/components/molecules/VenueRoomItem/VenueRoomItem.tsx]: Fixed in: https://github.com/sparkletown/sparkle/pull/2312/commits/0023b3b4d849d0af4ae51d5af3ca558495cb927e

**Review by goran-peoski-work — CHANGES_REQUESTED** [review]
Code looks good, apart from the need to merge latest changes from `staging` and replace user facing room/rooms instances with taxons https://github.com/sparkletown/sparkle/blob/staging/src/settings/taxonomy.ts

**Comment by denisdimitrov** [comment]
@goran-peoski-work Just made sure it's working properly and added taxons for the portals.

**Review by goran-peoski-work — APPROVED** [review]
LGTM. can be merged after this block from https://github.com/sparkletown/sparkle/blob/staging/src/scss/constants.scss#L151-L163 is safely merged back from staging

**Pull request merged**